hack/lib/test.sh

```diff
@@ -34,7 +34,7 @@ kube::test::get_object_assert() {
 
   res=$(eval kubectl ${args} get "${kube_flags[@]}" $object -o go-template=\"$request\")
 
-  if [[ "$res" =~ ^$expected$ ]]; then
+  if [[ "$res" =~ ^$expected ]]; then
```

Comment:
For example, 
```sh
  # Post-condition: valid-pod POD doesn't exist
  kube::test::get_object_assert pods "{{range.items}}{{$id_field}}:{{end}}" ''
```

If we remove the trailing `$` will the test pass when valid-pod POD exists?